tests/query/processors/test_prewhere.py

```diff
@@ -34,36 +57,122 @@
+                    (Column(None, "c", None), Literal(None, "3")),
+                ),
+            ),
+        ),
         [["a", "=", "1"], ["b", "=", "2"]],
+        FunctionCall(
+            None,
+            BooleanFunctions.AND,
+            (
+                FunctionCall(
+                    None,
+                    OPERATOR_TO_FUNCTION["="],
+                    (Column(None, "a", None), Literal(None, "1")),
+                ),
+                FunctionCall(
+                    None,
+                    OPERATOR_TO_FUNCTION["="],
+                    (Column(None, "b", None), Literal(None, "2")),
+                ),
+            ),
+        ),
     ),
     (
         # Do not add conditions that are parts of an OR
         {"conditions": [[["a", "=", "1"], ["b", "=", "2"]], ["c", "=", "3"]]},
         ["a", "b", "c"],
         [[["a", "=", "1"], ["b", "=", "2"]]],
+        FunctionCall(
+            None,
+            BooleanFunctions.OR,
+            (
+                FunctionCall(
+                    None,
+                    OPERATOR_TO_FUNCTION["="],
+                    (Column(None, "a", None), Literal(None, "1")),
+                ),
+                FunctionCall(
+                    None,
+                    OPERATOR_TO_FUNCTION["="],
+                    (Column(None, "b", None), Literal(None, "2")),
+                ),
+            ),
+        ),
         [["c", "=", "3"]],
+        FunctionCall(
+            None,
+            OPERATOR_TO_FUNCTION["="],
+            (Column(None, "c", None), Literal(None, "3")),
+        ),
     ),
     (
         # Exclude NOT IN condition from the prewhere as they are generally not excluding
         # most of the dataset.
         {"conditions": [["a", "NOT IN", [1, 2, 3]], ["b", "=", "2"], ["c", "=", "3"]]},
         ["a", "b"],
         [["a", "NOT IN", [1, 2, 3]], ["c", "=", "3"]],
+        FunctionCall(
+            None,
+            BooleanFunctions.AND,
+            (
+                not_in_condition(
+                    None,
+                    Column(None, "a", None),
+                    [Literal(None, 1), Literal(None, 2), Literal(None, 3)],
+                ),
+                FunctionCall(
+                    None,
+                    OPERATOR_TO_FUNCTION["="],
+                    (Column(None, "c", None), Literal(None, "3")),
+                ),
+            ),
+        ),
         [["b", "=", "2"]],
+        FunctionCall(
+            None,
+            OPERATOR_TO_FUNCTION["="],
+            (Column(None, "b", None), Literal(None, "2")),
+        ),
     ),
 ]
 
 
 @pytest.mark.parametrize(
-    "query_body, keys, new_conditions, prewhere_conditions", test_data
+    "query_body, keys, new_conditions, new_ast_condition, prewhere_conditions, new_prewhere_ast_condition",
+    test_data,
 )
-def test_prewhere(query_body, keys, new_conditions, prewhere_conditions) -> None:
+def test_prewhere(
+    query_body,
+    keys,
```

Comment:
Was keys also meant to have a type annotation?

Reply:
yes, thanks, I did not notice it. Will fix.